src/globals.ts

```diff
@@ -134,10 +140,35 @@ export const ExecutionTags = (() => {
   // @ts-ignore
   const clear = () => (global.tags = []);
 
+  const getValue = (autoTagEvent: AutoTagEvent, innerKey) => {
+    let obj = autoTagEvent.event;
+    const eventByKey = autoTagEvent.keyToEventMap;
+    if (eventByKey && eventByKey[innerKey]) {
+      obj = eventByKey[innerKey];
+      return obj && { event: obj, keyToEventMap: eventByKey };
+    }
+    try {
+      if (obj && isString(obj) && obj[innerKey] === undefined) {
+        const parsedObj = JSON.parse(obj);
+        eventByKey[innerKey] = parsedObj[innerKey];
```

Comment:
you use the inner key and not the full key to map the object and this can cause bugs for example in:
```
{
   a.duplicated_inner_key: "first_value",
   b.duplicated_inner_key: "second_value",
}
```